BillRunState.cpp

```diff
@@ -11,25 +10,58 @@ BillRunState::~BillRunState()
 
 void BillRunState::Exit(Bill& bill)
 {
-	bill.SetAnimation(BILL_RUN, bill.GetPosition(), direction);
 }
 
 void BillRunState::Enter(Bill& bill)
 {
-	bill.SetAnimation(BILL_RUN, bill.GetPosition(), direction);
+	if (bill.GetDirection() == LEFT)
+	{
+		bill.SetVX(+2.00f);
+		bill.SetAX(+0.01f);
+	}
+	if (bill.GetDirection() == RIGHT)
+	{
+		bill.SetVX(-2.00f);
+		bill.SetAX(-0.01f);
+	}
 }
 
 void BillRunState::Render(Bill& bill)
 {
-	bill.SetAnimation(BILL_RUN, bill.GetPosition(), direction);
+	bill.SetAnimation(BILL_RUN, bill.GetPosition(), bill.GetDirection());
 }
 
 BillState* BillRunState::Update(Bill& bill)
 {
-	if (direction == LEFT)
-		bill.SetX(bill.GetX() - bill.GetVX());
-	if (direction == RIGHT)
-		bill.SetX(bill.GetX() + bill.GetVX());
+	if (bill.GetDirection() == LEFT)
+	{
+		bill.SetVX(-abs(bill.GetVX()));
+		bill.SetAX(-abs(bill.GetAX()));
+	}
+	if (bill.GetDirection() == RIGHT)
+	{
+		bill.SetVX(+abs(bill.GetVX()));
+		bill.SetAX(+abs(bill.GetAX()));
+	}
+
+	bill.SetX
+	(
+		bill.GetX() + bill.GetVX() * time + bill.GetAX() * pow(time, 2) / 2
+	);
+
+	if (abs(bill.GetVX()) < +3.00f)
```

Comment:
why 3.00f

Reply:
Limit the Ox velocity => uniform motion when Ox velocity >= 3. Entity will move too fast if not.

Reply:
comment to code for other member understand

Reply:
I put the comment in code, please have a review.